### ISSUE TITLE
Bug: function-paren-newline error with jsx and multiple map

### ISSUE THREAD

**Issue by bertho-zero** [eslint/eslint#16777]
### Environment

Node version: 16.15.0
npm version: 9.2.0
Local ESLint version: 8.24.0
Global ESLint version: 8.31.0
Operating System: Ubuntu

### What parser are you using?

@babel/eslint-parser

### What did you do?

<details>
<summary>Configuration</summary>

```
"function-paren-newline": ["error","multiline-arguments"]
```
</details>

```js
function test() {
  return (
    <div>
      {pages?.map(page =>
        page.events.map(event => (
          <EventItem key={event.uid} event={event} />
        ))
      )}
    </div>
  );
}
```

### What did you expect to happen?

No error.

I would like the closing curly bracket to be at the same level as the opening curly bracket.

### What actually happened?

An error and a bad auto-fix:

```js
function test() {
  return (
    <div>
      {pages?.map(page =>
        page.events.map(event => (
          <EventItem key={event.uid} event={event} />
        )))}
    </div>
  );
}
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

The style is good for a map but not a map within a map because it's not directly jsx, I believe.

[Repro](https://eslint.org/play/#eyJ0ZXh0IjoiY29uc3QgcGFnZXMgPSBbXTtcblxuZXhwb3J0IGRlZmF1bHQgZnVuY3Rpb24gdGVzdCgpIHtcbiAgcmV0dXJuIChcbiAgICA8ZGl2PlxuICAgICAge3BhZ2VzPy5tYXAocGFnZSA9PlxuICAgICAgICBwYWdlLmV2ZW50cy5tYXAoZXZlbnQgPT4gKFxuICAgICAgICAgIDxFdmVudEl0ZW0ga2V5PXtldmVudC51aWR9IGV2ZW50PXtldmVudH0gLz5cbiAgICAgICAgKSlcbiAgICAgICl9XG4gICAgPC9kaXY+XG4gICk7XG59XG5cbmV4cG9ydCBmdW5jdGlvbiB0ZXN0MigpIHtcbiAgcmV0dXJuIChcbiAgICA8ZGl2PlxuICAgICAge3BhZ2VzPy5tYXAocGFnZSA9PiAoXG4gICAgICAgIDxFdmVudExpc3Qga2V5PXtwYWdlLmFmdGVyfSBldmVudHM9e3BhZ2UuZXZlbnR9IC8+XG4gICAgICApKX1cbiAgICA8L2Rpdj5cbiAgKTtcbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7ImpzeCI6dHJ1ZX19LCJydWxlcyI6eyJjb25zdHJ1Y3Rvci1zdXBlciI6WyJlcnJvciJdLCJmb3ItZGlyZWN0aW9uIjpbImVycm9yIl0sImdldHRlci1yZXR1cm4iOlsiZXJyb3IiXSwibm8tYXN5bmMtcHJvbWlzZS1leGVjdXRvciI6WyJlcnJvciJdLCJuby1jYXNlLWRlY2xhcmF0aW9ucyI6WyJlcnJvciJdLCJuby1jbGFzcy1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29tcGFyZS1uZWctemVybyI6WyJlcnJvciJdLCJuby1jb25kLWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb25zdC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29uc3RhbnQtY29uZGl0aW9uIjpbImVycm9yIl0sIm5vLWNvbnRyb2wtcmVnZXgiOlsiZXJyb3IiXSwibm8tZGVidWdnZXIiOlsiZXJyb3IiXSwibm8tZGVsZXRlLXZhciI6WyJlcnJvciJdLCJuby1kdXBlLWFyZ3MiOlsiZXJyb3IiXSwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjpbImVycm9yIl0sIm5vLWR1cGUtZWxzZS1pZiI6WyJlcnJvciJdLCJuby1kdXBlLWtleXMiOlsiZXJyb3IiXSwibm8tZHVwbGljYXRlLWNhc2UiOlsiZXJyb3IiXSwibm8tZW1wdHkiOlsiZXJyb3IiXSwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjpbImVycm9yIl0sIm5vLWVtcHR5LXBhdHRlcm4iOlsiZXJyb3IiXSwibm8tZXgtYXNzaWduIjpbImVycm9yIl0sIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6WyJlcnJvciJdLCJuby1leHRyYS1zZW1pIjpbImVycm9yIl0sIm5vLWZhbGx0aHJvdWdoIjpbImVycm9yIl0sIm5vLWZ1bmMtYXNzaWduIjpbImVycm9yIl0sIm5vLWdsb2JhbC1hc3NpZ24iOlsiZXJyb3IiXSwibm8taW1wb3J0LWFzc2lnbiI6WyJlcnJvciJdLCJuby1pbm5lci1kZWNsYXJhdGlvbnMiOlsiZXJyb3IiXSwibm8taW52YWxpZC1yZWdleHAiOlsiZXJyb3IiXSwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOlsiZXJyb3IiXSwibm8tbG9zcy1vZi1wcmVjaXNpb24iOlsiZXJyb3IiXSwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOlsiZXJyb3IiXSwibm8tbWl4ZWQtc3BhY2VzLWFuZC10YWJzIjpbImVycm9yIl0sIm5vLW5ldy1zeW1ib2wiOlsiZXJyb3IiXSwibm8tbm9ub2N0YWwtZGVjaW1hbC1lc2NhcGUiOlsiZXJyb3IiXSwibm8tb2JqLWNhbGxzIjpbImVycm9yIl0sIm5vLW9jdGFsIjpbImVycm9yIl0sIm5vLXByb3RvdHlwZS1idWlsdGlucyI6WyJlcnJvciJdLCJuby1yZWRlY2xhcmUiOlsiZXJyb3IiXSwibm8tcmVnZXgtc3BhY2VzIjpbImVycm9yIl0sIm5vLXNlbGYtYXNzaWduIjpbImVycm9yIl0sIm5vLXNldHRlci1yZXR1cm4iOlsiZXJyb3IiXSwibm8tc2hhZG93LXJlc3RyaWN0ZWQtbmFtZXMiOlsiZXJyb3IiXSwibm8tc3BhcnNlLWFycmF5cyI6WyJlcnJvciJdLCJuby10aGlzLWJlZm9yZS1zdXBlciI6WyJlcnJvciJdLCJuby11bmRlZiI6WyJlcnJvciJdLCJuby11bmV4cGVjdGVkLW11bHRpbGluZSI6WyJlcnJvciJdLCJuby11bnJlYWNoYWJsZSI6WyJlcnJvciJdLCJuby11bnNhZmUtZmluYWxseSI6WyJlcnJvciJdLCJuby11bnNhZmUtbmVnYXRpb24iOlsiZXJyb3IiXSwibm8tdW5zYWZlLW9wdGlvbmFsLWNoYWluaW5nIjpbImVycm9yIl0sIm5vLXVudXNlZC1sYWJlbHMiOlsiZXJyb3IiXSwibm8tdW51c2VkLXZhcnMiOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1iYWNrcmVmZXJlbmNlIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtY2F0Y2giOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1lc2NhcGUiOlsiZXJyb3IiXSwibm8td2l0aCI6WyJlcnJvciJdLCJyZXF1aXJlLXlpZWxkIjpbImVycm9yIl0sInVzZS1pc25hbiI6WyJlcnJvciJdLCJ2YWxpZC10eXBlb2YiOlsiZXJyb3IiXSwiZnVuY3Rpb24tcGFyZW4tbmV3bGluZSI6WyJlcnJvciIsIm11bHRpbGluZS1hcmd1bWVudHMiXX0sImVudiI6eyJicm93c2VyIjp0cnVlfX19)

**Comment by mdjermanovic** [comment]
Hi @bertho-zero, thanks for the issue!

This works as intended because this rule always enforces consistency between newlines after `(` and before `)` of the same function call: there should be both or none. In your example, there should be either newlines both after `pages?.map(` and before its `)`, or no newlines both after `pages?.map(` and before its `)`.

> ```js
> function test() {
>   return (
>     <div>
>       {pages?.map(page =>
>         page.events.map(event => (
>           <EventItem key={event.uid} event={event} />
>         ))
>       )}
>     </div>
>   );
> }
> ```

The above code is invalid for this rule because there's no newline after `pages?.map(` but there is a newline before `)`.

> ```js
> function test() {
>   return (
>     <div>
>       {pages?.map(page =>
>         page.events.map(event => (
>           <EventItem key={event.uid} event={event} />
>         )))}
>     </div>
>   );
> }
> ```

The above code is valid for this rule because there's no newline after `pages?.map(` and there's no newline before `)`.

As an alternative to this formatting, the following would also be valid for this rule as there are newlines both after `pages?.map(` and before `)`:

```js
function test() {
  return (
    <div>
      {pages?.map(
        page =>
          page.events.map(event => (
            <EventItem key={event.uid} event={event} />
          ))
      )}
    </div>
  );
}
```

**Comment by bertho-zero** [comment]
Thanks for your response, @mdjermanovic
